components/remote-environment-broker/internal/syncer/ns_broker_syncer_test.go

```diff
@@ -0,0 +1,113 @@
+package syncer
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kyma-project/kyma/components/remote-environment-broker/internal/syncer/automock"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestServiceBrokerSync_Sync_Success(t *testing.T) {
+	// given
+	labelSelector := "app=label"
+
+	syncer := &automock.ServiceBrokerSyncer{}
+	syncer.On("List", v1.NamespaceAll, v1.ListOptions{
+		LabelSelector: labelSelector,
+	}).Return(fixServiceBrokersList(), nil)
+	syncer.On("Update", fixServiceBroker(1)).Return(fixServiceBroker(1), nil)
+	defer syncer.AssertExpectations(t)
+
+	nsBrokerSync := NewServiceBrokerSyncer(syncer)
+
+	// when
+	err := nsBrokerSync.Sync(labelSelector, maxSyncRetries)
+
+	// then
+	require.NoError(t, err)
+}
+
+func TestServiceBrokerSync_Sync_Empty(t *testing.T) {
+	// given
+	labelSelector := "app=label"
+
+	syncer := &automock.ServiceBrokerSyncer{}
+	syncer.On("List", v1.NamespaceAll, v1.ListOptions{
+		LabelSelector: labelSelector,
+	}).Return(&v1beta1.ServiceBrokerList{}, nil)
+	defer syncer.AssertExpectations(t)
+
+	nsBrokerSync := NewServiceBrokerSyncer(syncer)
+
+	// when
+	err := nsBrokerSync.Sync(labelSelector, maxSyncRetries)
+
+	// then
+	require.NoError(t, err)
+}
+
+func TestServiceBrokerSync_Sync_Error(t *testing.T) {
+	// given
+	labelSelector := "app=label"
+
+	syncer := &automock.ServiceBrokerSyncer{}
+	syncer.On("List", v1.NamespaceAll, v1.ListOptions{
+		LabelSelector: labelSelector,
+	}).Return(fixServiceBrokersList(), nil)
+	syncer.On("Update", fixServiceBroker(1)).Return(fixServiceBroker(1), fmt.Errorf("fix"))
+	defer syncer.AssertExpectations(t)
+
+	nsBrokerSync := NewServiceBrokerSyncer(syncer)
+
+	// when
+	err := nsBrokerSync.Sync(labelSelector, 1)
+
+	// then
+	require.Error(t, err)
+}
+
+func TestServiceBrokerSync_Sync_MultiError(t *testing.T) {
+	// given
+	labelSelector := "app=label"
+	maxRetries := 5
+
+	syncer := &automock.ServiceBrokerSyncer{}
+	syncer.On("List", v1.NamespaceAll, v1.ListOptions{
+		LabelSelector: labelSelector,
+	}).Return(fixServiceBrokersList(), nil)
+	syncer.On("Update", fixServiceBroker(1)).
+		Return(fixServiceBroker(1), fmt.Errorf("fix")).Times(maxRetries)
+	defer syncer.AssertExpectations(t)
+
+	nsBrokerSync := NewServiceBrokerSyncer(syncer)
+
+	// when
+	err := nsBrokerSync.Sync(labelSelector, maxRetries)
+
+	// then
+	require.Error(t, err)
+}
+
+func fixServiceBroker(relistRequests int64) *v1beta1.ServiceBroker {
```

Comment:
for me it is not clear what 1 means when I see:
fixServiceBroker(1), change it to fixServiceBrokerWithRelistReq or sth similar